azurerm/data_source_front_door.go

```diff
@@ -0,0 +1,392 @@
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"host_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"session_affinity_enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"session_affinity_ttl_seconds": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"custom_https_configuration": {
+							Type:     schema.TypeList,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"certificate_source": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									// NOTE: None of these attributes are valid if
+									//       certificate_source is set to FrontDoor
+									"azure_key_vault_certificate_secret_name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"azure_key_vault_certificate_secret_version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"azure_key_vault_certificate_vault_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"friendly_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceArmFrontDoorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).frontDoorsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["frontdoors"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Front Door %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Front Door %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if properties := resp.Properties; properties != nil {
+		if err := d.Set("backend_pool", flattenArmFrontDoorBackendPools(properties.BackendPools)); err != nil {
+			return fmt.Errorf("Error setting `backend_pool`: %+v", err)
+		}
+		if err := d.Set("enforce_backend_pools_certificate_name_check", flattenArmFrontDoorBackendPoolsSettings(properties.BackendPoolsSettings)); err != nil {
+			return fmt.Errorf("Error setting `enforce_backend_pools_certificate_name_check`: %+v", err)
+		}
+		d.Set("cname", properties.Cname)
+		if properties.EnabledState == frontdoor.EnabledStateEnabled {
+			d.Set("enabled", true)
```

Comment:
I think this could be:
```suggestion
			d.Set("enabled", properties.EnabledState == frontdoor.EnabledStateEnabled)
```